### ISSUE TITLE
MOSAIC on OBJs still implemented wrong

### ISSUE THREAD

**Issue by endrift** [visualboyadvance-m/visualboyadvance-m#785]
### Expected behavior

[First reported (by me at least) over 8 years ago](https://board.vba-m.com/viewtopic.php?f=17&t=240), OBJ mosaic has visible differences from hardware. At the _very least_, the mosaic grid is supposed to be globally aligned, not locally aligned, but the fine details are much more finicky.

### Actual behavior

![](http://i.endrift.com/2/mosaic.png)

### Steps to reproduce the behavior

This is an example from the intro of Aria of Sorrow, but there are a ton of other examples.

### ROM details

Which games are affected by the issue (please state if GBA or GB game): Every game that uses OBJ mosaic

Which games, if any, are NOT affected by the issue (please state if GBA or GB game): Games that don't use OBJ mosaic

### Option details

Video driver (simple, opengl): simple (doesn't matter though)

Sound driver (directsound, xaudio, openal, SDL): N/A

### Build details

Operating System (Windows, Mac, Linux (state distribution), etc.): N/A

If using Linux, specify if using xorg or Wayland: N/A 

Version of code (pre-built binary version, or commit ref, or just "master"): dfa97d8a916fddd4a0fe0b34a07c44abec29d78a

**Comment by andyvand** [comment]
This works now fine

**Comment by Squall-Leonhart** [comment]
No, our mosaics are off in general, we're far from accurate on both [HMosaic](https://github.com/nba-emu/hw-test/tree/master/ppu/sprite-hmosaic) and [VMosaic](https://github.com/mgba-emu/mgba/issues/1008).

**Comment by Squall-Leonhart** [comment]
Now we do them right (or wrong good enough to fool frame comparisons)
